src/app/routes/getInitialData/index.js

```diff
@@ -0,0 +1,52 @@
+import 'isomorphic-fetch';
+import isAmpPath from '../../helpers/isAmpPath';
+import isServer from '../../helpers/isServer';
+
+const validateService = service => {
+  const services = ['news', 'persian'];
+  const serviceMatch = services.includes(service);
+
+  if (!serviceMatch) {
+    throw new Error(
+      `Invalid route parameter: ${service}. Service parameter must be news or persian.`,
+    );
+  }
+};
+
+const validateId = id => {
+  const regex = '^(c[a-zA-Z0-9]{10}o)$';
+  const routeMatches = id.match(regex);
+
+  if (!routeMatches) {
+    throw new Error(
+      `Invalid route parameter: ${id}. ID parameter must be in format 'c[xxxxxxxxxx]o', where the middle part could be 0000000001 to 0000000027.`,
+    );
+  }
+};
+
+const getInitialData = async ({ match }) => {
+  try {
+    const { path } = match;
+    const { id, service } = match.params;
+
+    validateService(service);
+    validateId(id);
+
+    let url = `/data/${service}/${id}.json`;
+
+    if (isServer()) {
+      url = `${process.env.RAZZLE_BASE_PATH}${url}`;
+    }
+
+    const response = await fetch(url);
+    const data = await response.json();
+    const amp = isAmpPath(path);
+
+    return { amp, data };
+  } catch (error) {
+    console.log(error); // eslint-disable-line no-console
+    return {};
```

Comment:
We need to refactor this to throw the error, so it can be caught [in the component](https://github.com/bbc/simorgh/pull/640/files#diff-d5c925f82e0315a863ea6a24dd59cb60R11), or [on the server](https://github.com/bbc/simorgh/pull/640/files#diff-788dcef7a5e3a2740bf71979cca64a59R57), but I did not want to change the underlying implementation.